src/shared/components/mutationTable/MutationTable.tsx

```diff
@@ -0,0 +1,340 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import {observable, computed} from "mobx";
+import * as _ from "lodash";
+import {default as MSKTable, Column} from "shared/components/msktable/MSKTable";
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import DiscreteCNAColumnFormatter from "./column/DiscreteCNAColumnFormatter";
+import AlleleCountColumnFormatter from "./column/AlleleCountColumnFormatter";
+import GeneColumnFormatter from "./column/GeneColumnFormatter";
+import ChromosomeColumnFormatter from "./column/ChromosomeColumnFormatter";
+import ProteinChangeColumnFormatter from "./column/ProteinChangeColumnFormatter";
+import MutationTypeColumnFormatter from "./column/MutationTypeColumnFormatter";
+import MutationAssessorColumnFormatter from "./column/MutationAssessorColumnFormatter";
+import CosmicColumnFormatter from "./column/CosmicColumnFormatter";
+import {ICosmicData} from "shared/model/Cosmic";
+import AnnotationColumnFormatter from "./column/AnnotationColumnFormatter";
+import {IMyCancerGenomeData} from "shared/model/MyCancerGenome";
+import {IHotspotData} from "shared/model/CancerHotspots";
+import {IOncoKbData} from "shared/model/OncoKB";
+import {IMutSigData} from "shared/model/MutSig";
+import DiscreteCNACache from "shared/cache/DiscreteCNACache";
+import OncoKbEvidenceCache from "shared/cache/OncoKbEvidenceCache";
+import PmidCache from "shared/cache/PmidCache";
+
+export interface IMutationTableProps {
+    sampleIds?:string[];
+    discreteCNACache?:DiscreteCNACache;
+    oncoKbEvidenceCache?:OncoKbEvidenceCache;
+    pmidCache?:PmidCache
+    mutSigData?:IMutSigData;
+    myCancerGenomeData?: IMyCancerGenomeData;
+    hotspots?: IHotspotData;
+    cosmicData?:ICosmicData;
+    oncoKbData?:IOncoKbData;
+    mrnaExprRankGeneticProfileId?:string;
+    discreteCNAGeneticProfileId?:string;
+    columns:MutationTableColumnType[];
+    data:Mutation[][];
+}
+
+export enum MutationTableColumnType {
```

Comment:
that's okay, I think that still makes sense. The base mutation table should implement all the columns, and each use of it should just specify which ones and the order